client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/index.tsx

```diff
@@ -19,10 +20,14 @@ const Launchpad: Step = ( { navigation }: LaunchpadProps ) => {
 	const translate = useTranslate();
 	const almostReadyToLaunchText = translate( 'Almost ready to launch' );
 	const siteSlug = useSiteSlugParam();
-
+	const flow = useFlowParam();
 	const site = useSite();
 	const launchpadScreenOption = site?.options?.launchpad_screen;
 
+	useEffect( () => {
+		recordTracksEvent( 'calypso_launchpad_loaded', { flow: flow } );
+	}, [ flow ] );
```

Comment:
I think this might be better in the useEffect below, so that we can only send the event if we aren't redirecting.

That also makes me wonder if a similar event would be useful when we do redirect, it seems like this could be useful in the future as well!

Reply:
@Addison-Stavlo The only issue with that is the `useEffect` gets called multiple times and will dispatch multiple duplicate tracks events (Unless I"m missing something)

![image](https://user-images.githubusercontent.com/20927667/193137501-dedc1409-ed09-41fd-abaf-e05a3e213fcc.png)

![image](https://user-images.githubusercontent.com/20927667/193137345-1013f153-1ca4-4767-a6b0-e015868b8fab.png)

Reply:
Interesting! I guess a question there is which dependencies are changing to cause this to be retriggered. And is it just something initializing from `null|undefined` to its expected value and not retriggering the component after that?

Maybe we wait to send the event until that value is initialized as expected by putting the event behind a conditional? 🤔

Reply:
@Addison-Stavlo It's the `launchpadScreenOption` changing from `undefined` to `off` or `full`. I guess we can do 
```
if ( launchpadScreenOption ) {
	recordTracksEvent( 'calypso_launchpad_loaded', { flow: flow } );
}
```

but it seems like a workaround because the `launchpadScreenOption` is not a dependency of the `recordTracksEvent` action

Reply:
It does seem like a bit of a workaround on the surface, doesn't it? Generally we would like to avoid work arounds as well and address the root of the problem.

But thinking about it more, I think it makes sense here. While the function itself doesn't rely on these other variables, the conditions in which we want to run it do: we only want to run the function to run when we know we aren't redirecting, and we only know we aren't redirecting after these other variables are initialized. So while the function may not depend on that variable for it's internal operation, it does depend on it for running only at the appropriate time (after we are sure we aren't redirecting).

It would be good to add a comment as to why we are doing it, as you point out it does not seem self explanatory and looks like an odd workaround.